File: FE/WEB/app/models/erp-site-user-mapping.js
import Model, { attr, belongsTo } from '@ember-data/model';

export default class ErpSiteUserMappingModel extends Model {
  @belongsTo('erp-site', { async: true, inverse: null }) site; // Relazione con la sede
  @belongsTo('user', { async: true, inverse: null }) user; // Relazione con l'utente

  @attr('number') tenantId; // Identificativo numerico del tenant
  @attr('string') siteId; // Identificativo della sede
  @attr('string') employeeId; // Identificativo del dipendente
  @attr('boolean', { defaultValue: false }) isPrimary; // Indica se è la sede principale per l’utente
  @attr('date-utc', {
    defaultValue() {
      return new Date();
    },
  })
  mappingStartDate; // Data di inizio del collegamento
  @attr('date-utc') mappingEndDate; // Data di fine del collegamento
  @attr('date-utc') createdAt; // Data di creazione del record
  @attr('date-utc', {
    defaultValue() {
      return new Date();
    },
  })
  updatedAt; // Data di ultima modifica del record
}

File: BE/Src/MIT.Fwk.Infrastructure/Entities/ErpSiteUserMapping.cs
using JsonApiDotNetCore.Resources;
using JsonApiDotNetCore.Resources.Annotations;
using System;
using System.ComponentModel.DataAnnotations.Schema;

namespace MIT.Fwk.Infrastructure.Entities
{

    [Table("ErpSiteUserMapping")]
    public class ErpSiteUserMapping : Identifiable<string>
    {
        [Attr] public int TenantId { get; set; }
        [Attr] public string SiteId { get; set; }
        [Attr] public string EmployeeId { get; set; }
        [Attr] public bool IsPrimary { get; set; }
        [Attr] public DateTime MappingStartDate { get; set; }
        [Attr] public DateTime? MappingEndDate { get; set; }
        [Attr] public DateTime CreatedAt { get; set; }
        [Attr] public DateTime UpdatedAt { get; set; }

        [HasOne] public virtual ErpSite Site { get; set; }
        [HasOne] public virtual User User { get; set; }
        [HasOne] public virtual Tenant Tenant { get; set; }
    }
}
